widgets/text/text.go

```diff
@@ -108,6 +109,25 @@ func (t *Text) Write(text string, wOpts ...WriteOption) error {
 	if opts.replace {
 		t.reset()
 	}
+
+	// If set, limit the buffer (if maxContent has been set)
```

Comment:
This latest version looks great, thank you for iterating on it. I wonder if we could make the Write function more readable by extracting all code related to content truncation into its own method. Something like:

```go
// Truncates the content and the new text so that the configured MaxContentCells
// is enforced after a Write. 
func (t *Text) truncateToMaxCells(text string) { ... }
```

The `Write` method would then just conditionally call the `truncateToMaxCells` method if `opts.maxContentCells` was set. This might make the "business logic" of `Write` easier to follow.

```go
if t.opts.maxContent > 0 {
  t.truncateToMaxCells(text)
}
```

Reply:
Happy for this to be moved out. Happy if you wanted to make these changes, the incoming `text` also needs to be truncated if required as well.